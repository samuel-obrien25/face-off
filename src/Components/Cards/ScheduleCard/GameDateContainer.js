import React from 'react';
import styled from 'styled-components';

const StyledGameDateContainer = styled.div`
    position: absolute;
    top:0;
    width: 100%;
    left:0;
    right:0;
    margin:auto;
    background-color: #222;
    color: #fff;
    height: 1.2em;
    padding-top: 5px;
    display: flex;
`;

const StyledGameDate = styled.p`
    margin: auto;
`;


const GameDateContainer = (props) => {

        //Gets date from API response and converts it to a readable format for the Card
        //Needs to get split off into its own component
        let jsonDate = new Date(props.gameDateTime),
            weekday = new Array(7);
                weekday[0] = "Sunday";
                weekday[1] = "Monday";
                weekday[2] = "Tuesday";
                weekday[3] = "Wednesday";
                weekday[4] = "Thursday";
                weekday[5] = "Friday";
                weekday[6] = "Saturday";

        let formattedGameDate = weekday[jsonDate.getDay()] + ', ' + jsonDate.toLocaleDateString() + ' at ' + jsonDate.toLocaleTimeString();

  return (
        <StyledGameDateContainer className={`team${props.activeTeamID}`}>
            <StyledGameDate> {formattedGameDate}</StyledGameDate>
        </StyledGameDateContainer>
  )

}

export default GameDateContainer;

