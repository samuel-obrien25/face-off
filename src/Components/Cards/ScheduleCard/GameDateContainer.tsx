import React from 'react';
import styled from 'styled-components';

const StyledGameDateContainer = styled.div`
    position: absolute;
    top:0;
    width: 100%;
    left:0;
    right:0;
    margin:auto;
    color: #fff;
    height: 1.2em;
    padding-top: 5px;
    display: flex;
`;

const StyledGameDate = styled.p`
    margin: auto;
`;

interface GameDateContainerProps {
    gameDateTime: Date,
    activeTeamID: string,
}

const GameDateContainer: React.FC<GameDateContainerProps> = (props) => {
    const { activeTeamID, gameDateTime} = props;

        //Gets date from API response and converts it to a readable format for the Card
        //Needs to get split off into its own component
        let jsonDate = new Date(gameDateTime),
            weekday = new Array(7);
                weekday[0] = "Sunday";
                weekday[1] = "Monday";
                weekday[2] = "Tuesday";
                weekday[3] = "Wednesday";
                weekday[4] = "Thursday";
                weekday[5] = "Friday";
                weekday[6] = "Saturday";

        let formattedGameDate = weekday[jsonDate.getDay()] + ', ' + jsonDate.toLocaleDateString() + ' at ' + jsonDate.toLocaleTimeString();

  return (
        <StyledGameDateContainer gameDateTime={props.gameDateTime} className={`team${activeTeamID}`} {...props}>
            <StyledGameDate> {formattedGameDate}</StyledGameDate>
        </StyledGameDateContainer>
  )

}

export default GameDateContainer;

