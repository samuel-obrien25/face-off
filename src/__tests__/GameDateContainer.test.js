import React from 'react';
import { shallow, mount, render } from 'enzyme';

import GameDateContainer from '../Components/Cards/ScheduleCard/GameDateContainer';

describe('GameDateContainer', () => {
//It's defined
    it('should be defined', () => {
        expect(GameDateContainer).toBeDefined();
    });

//It renders correctly
    it('Should render correctly', () => {
        const mockDate = "Test Date";
        const component = shallow(
            <GameDateContainer>{mockDate}</GameDateContainer>
        );
        expect(component).toMatchSnapshot();
    });
});