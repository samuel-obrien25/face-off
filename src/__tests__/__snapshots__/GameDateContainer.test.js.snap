// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameDateContainer Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GameDateContainer>
    Test Date
  </GameDateContainer>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef>
         
        undefined, Invalid Date at Invalid Date
      </ForwardRef>,
      "className": "teamundefined",
      "gameDateTime": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          " ",
          "undefined, Invalid Date at Invalid Date",
        ],
      },
      "ref": null,
      "rendered": Array [
        " ",
        "undefined, Invalid Date at Invalid Date",
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
    margin: auto;
",
          ],
        },
        "displayName": "styled.p",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "p",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    position: absolute;
    top:0;
    width: 100%;
    left:0;
    right:0;
    margin:auto;
    color: #fff;
    height: 1.2em;
    padding-top: 5px;
    display: flex;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
           
          undefined, Invalid Date at Invalid Date
        </ForwardRef>,
        "className": "teamundefined",
        "gameDateTime": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            " ",
            "undefined, Invalid Date at Invalid Date",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          "undefined, Invalid Date at Invalid Date",
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
    margin: auto;
",
            ],
          },
          "displayName": "styled.p",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "p",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    position: absolute;
    top:0;
    width: 100%;
    left:0;
    right:0;
    margin:auto;
    color: #fff;
    height: 1.2em;
    padding-top: 5px;
    display: flex;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
